naga/src/front/wgsl/mod.rs

```diff
@@ -44,6 +44,17 @@ impl Frontend {
     }
 }
 
+/// <div class="warning">
+// NOTE: Keep this in sync with `wgpu::Device::create_shader_module`!
+// NOTE: Keep this in sync with `wgpu_core::Global::device_create_shader_module`!
+///
+/// Compiler-enforced limits for WGSL parsing recursion exist; if shader compilation runs into
+/// them, it will return an error gracefully. However, on some build profiles and platforms, the
+/// default stack size for a thread may be exceeded before this limit is reached during parsing.
+/// Callers should ensure that there is enough stack space for this, particularly if calls to this
+/// method are exposed to user input.
```

Comment:
This comment buries the lede. It should say:

> Note that this function may consume quite a lot of stack space. Compiler-enforced limits ...

Reply:
Oh, I thought the `warning` box was sufficient encouragement to read the whole thing, but I have no objection to actually stating the key takeaway right off. 🙂 Will change as suggested shortly.

Reply:
Fixed up with 8cdd967ca.